torch/nn/parallel/scatter_gather.py

```diff
@@ -1,25 +1,36 @@
+import torch
 from torch.autograd import Variable
 from ._functions import Scatter, Gather
+from torch.cuda.comm import broadcast
 
 
-def scatter(input, target_gpus):
-    """Slices a given variable into approximately equal chunks and distributes
-       them accross given GPUs
+def scatter(input, target_gpus, dim=0):
+    """
+    Slices a given variable into approximately equal chunks and distributes
+      them accross given GPUs
+    Duplicates references to objects that are not variables.
     """
     def scatter_map(obj):
         if isinstance(obj, Variable):
-            return Scatter(target_gpus)(obj)
-        return tuple(zip(*map(scatter_map, obj)))
+            return Scatter(target_gpus, dim=dim)(obj)
+        if isinstance(obj, tuple) or isinstance(obj, list):
+            return type(obj)(zip(*map(scatter_map, obj)))
+        if torch.is_tensor(obj):
+            return broadcast(obj, target_gpus)
```

Comment:
I haven't done kwargs yet, but was thinking. A good compromise may be to scatter all variables and tensors that come in as normal positional arguments and broadcast all variables and tensors that come in as keyword arguments. May do the second as a separate PR, as it requires processing the dict of kwargs and each entry as the inputs are now.